tracee-ebpf/embed-btfhub.go

```diff
@@ -0,0 +1,14 @@
+//go:build btfhub
+// +build btfhub
+
+package main
+
+import (
+	"embed"
+)
+
+var btfEmbedded bool = true
+
+//go:embed "dist/btfhub/*"
```

Comment:
I actually think we should embed this in the executable itself, and not as part of the container. The embedded size is small enough to be part of the binary. This way, a user can run tracee-ebpf as a standalone binary, without having to provide an external btf file.

Reply:
I liked very much the embedded approach, to be honest. Outcome has accelerated development (an example showed here: https://github.com/aquasecurity/tracee/pull/1130#pullrequestreview-802126454). 

Despite tracee being a security *solution*, it may also be used as an OS application and, in that case, the embedded approach makes users lives much easier. I think that could help in adoption (considering we package tracee-ebpf and tracee-rules as OS packages, like we planed) for LTS Linux users. 

*Note*: FYI, Showing the embedded approach today also called attention of CNDR team...

Reply:
After talking today with @rafaeldtinoco, it's actually not so critical for us whether the BTFs are embedded or not. Anyways the BPF CORE object is aready embedded and we are still managing to 'go get' tracee ('go get' -d).

Reply:
@itaysk @yanivagman 

I have pushed an example on how btfhub can be integrated with our makefile without submodules: through a script that can be used during compilation. If BTFHUB=1 then the script is responsible for dealing with the BTFhub repositories and logic.

After sometime "fighting" with our Makefile targets, I think we should re-work our Makefile, splitting docker tracee-builder, docker container generation AND the regular build targets. 

> I have the impression that some build targets are being built twice because of target dependencies made to PHONY targets (and things alike).

Anything against re-working Makefile entirely ? This way I can also fix:

```make
.PHONY: all
all:
        $(MAKE) bpf
        $(MAKE) btfhub
        $(MAKE) build
```

making the correct target dependencies.

Nevertheless I need a decision on embedding BTF files in golang binary (which I'm in favor of) or not. WDYT ?

Reply:
We've agreed on the Makefile improvement (in next PR) and to have embedded BTF files within the golang binary (not the best approach ever, but "acceptable").